drivers/include/periph/led.h

```diff
@@ -0,0 +1,146 @@
+#define LED1_ON                 gpio_set(LED1_PIN)
+#define LED1_OFF                gpio_clear(LED1_PIN)
+#define LED1_TOGGLE             gpio_toggle(LED1_PIN)
+#endif /* LED1_ON */
+#ifndef LED1_INIT
+#define LED1_INIT               do { gpio_init(LED1_PIN, GPIO_OUT); LED1_OFF; } while(0)
+#endif /* LED1_INIT */
+#endif /* LED1_PIN */
+
+#ifdef LED2_PIN
+#ifndef LED2_ON
+/* assuming LED2_OFF and LED2_TOGGLE are undefined too */
+#define LED2_ON                 gpio_set(LED2_PIN)
+#define LED2_OFF                gpio_clear(LED2_PIN)
+#define LED2_TOGGLE             gpio_toggle(LED2_PIN)
+#endif /* LED2_ON */
+#ifndef LED2_INIT
+#define LED2_INIT               do { gpio_init(LED2_PIN, GPIO_OUT); LED2_OFF; } while(0)
+#endif /* LED2_INIT */
+#endif /* LED2_PIN */
+
+#ifdef LED3_PIN
+#ifndef LED3_ON
+/* assuming LED3_OFF and LED3_TOGGLE are undefined too */
+#define LED3_ON                 gpio_set(LED3_PIN)
+#define LED3_OFF                gpio_clear(LED3_PIN)
+#define LED3_TOGGLE             gpio_toggle(LED3_PIN)
+#endif /* LED3_ON */
+#ifndef LED3_INIT
+#define LED3_INIT               do { gpio_init(LED3_PIN, GPIO_OUT); LED3_OFF; } while(0)
+#endif /* LED3_INIT */
+#endif /* LED3_PIN */
+
+#ifdef LED4_PIN
+#ifndef LED4_ON
+/* assuming LED4_OFF and LED4_TOGGLE are undefined too */
+#define LED4_ON                 gpio_set(LED4_PIN)
+#define LED4_OFF                gpio_clear(LED4_PIN)
+#define LED4_TOGGLE             gpio_toggle(LED4_PIN)
+#endif /* LED4_ON */
+#ifndef LED4_INIT
+#define LED4_INIT               do { gpio_init(LED4_PIN, GPIO_OUT); LED4_OFF; } while(0)
+#endif /* LED4_INIT */
+#endif /* LED4_PIN */
+
+#ifdef LED5_PIN
+#ifndef LED5_ON
+/* assuming LED5_OFF and LED5_TOGGLE are undefined too */
+#define LED5_ON                 gpio_set(LED5_PIN)
+#define LED5_OFF                gpio_clear(LED5_PIN)
+#define LED5_TOGGLE             gpio_toggle(LED5_PIN)
+#endif /* LED5_ON */
+#ifndef LED5_INIT
+#define LED5_INIT               do { gpio_init(LED5_PIN, GPIO_OUT); LED5_OFF; } while(0)
+#endif /* LED5_INIT */
+#endif /* LED5_PIN */
+
+#ifdef LED6_PIN
+#ifndef LED6_ON
+/* assuming LED6_OFF and LED6_TOGGLE are undefined too */
+#define LED6_ON                 gpio_set(LED6_PIN)
+#define LED6_OFF                gpio_clear(LED6_PIN)
+#define LED6_TOGGLE             gpio_toggle(LED6_PIN)
+#endif /* LED6_ON */
+#ifndef LED6_INIT
+#define LED6_INIT               do { gpio_init(LED6_PIN, GPIO_OUT); LED6_OFF; } while(0)
+#endif /* LED6_INIT */
+#endif /* LED6_PIN */
+
+#ifdef LED7_PIN
+#ifndef LED7_ON
+/* assuming LED7_OFF and LED7_TOGGLE are undefined too */
+#define LED7_ON                 gpio_set(LED7_PIN)
+#define LED7_OFF                gpio_clear(LED7_PIN)
+#define LED7_TOGGLE             gpio_toggle(LED7_PIN)
+#endif /* LED7_ON */
+#ifndef LED7_INIT
+#define LED7_INIT               do { gpio_init(LED7_PIN, GPIO_OUT); LED7_OFF; } while(0)
+#endif /* LED7_INIT */
+#endif /* LED7_PIN */
+/** @} */
```

Comment:
Maybe instead of creating a define per "possibly" LEDs present in a board (what if I have 8 LEDs? This won't work), you can propose a definition similar to UART, SPI or (future) I2C, on which you have a general structure and is the board which defines how many "objects" of this kind it has, this way you'll avoid the need to define everything possible.

However, I doubt if such a solution would have a negative impact on code size.